clients/githubrepo/branches.go

```diff
@@ -358,39 +358,36 @@ func copyNonAdminSettings(src interface{}, dst *clients.BranchProtectionRule) {
 		copyBoolPtr(v.AllowsDeletions, &dst.AllowDeletions)
 		copyBoolPtr(v.AllowsForcePushes, &dst.AllowForcePushes)
 		copyBoolPtr(v.RequiresLinearHistory, &dst.RequireLinearHistory)
+		copyInt32Ptr(v.RequiredApprovingReviewCount, &dst.RequiredPullRequestReviews.RequiredApprovingReviewCount)
+		copyBoolPtr(v.RequiresCodeOwnerReviews, &dst.RequiredPullRequestReviews.RequireCodeOwnerReviews)
 		copyStringSlice(v.RequiredStatusCheckContexts, &dst.CheckRules.Contexts)
 
-		// If branch doesn't require PRs to make changes, we let the struct RequiredPullRequestReviews point to nil
-		if branchRequiresPrs(v) {
-			if dst.RequiredPullRequestReviews == nil {
-				dst.RequiredPullRequestReviews = new(clients.PullRequestReviewRule)
-			}
-			copyInt32Ptr(v.RequiredApprovingReviewCount, &dst.RequiredPullRequestReviews.RequiredApprovingReviewCount)
-			copyBoolPtr(v.RequiresCodeOwnerReviews, &dst.RequiredPullRequestReviews.RequireCodeOwnerReviews)
+		// we always have the data to know if PRs are required
+		if dst.RequiredPullRequestReviews.Required == nil {
+			dst.RequiredPullRequestReviews.Required = new(bool)
+		}
+		// RequiredApprovingReviewCount is nil if PRs aren't required, or &num if PRs are required (including &0)
+		// RequiresCodeOwnerReviews is &false even if PRs aren't required, so we need it to be true
```

Comment:
b6c7bff54520977b15b56dcde2ea75feddc42758

Reply:
Oops, I was actually referring to the second line ><
I didn't understand the "so we need it to be true". Because we don't actually need the requireCodeOwnerReviews to be true. We just use a possible "true" value to imply that PRs are required.

Reply:
GitHub reviews are misleading if you select exactly 1 line. So I try to pick at least 2 if it helps avoid ambiguity